File: apps/web/src/app/core/api/api-headers.ts
/**
 * Custom HTTP headers that are part of the API contract.
 *
 * Mirrors `TouchlineManager.Contracts.Http.ApiHeaders` on the server. Later stages generate the
 * DTO types from the committed OpenAPI document; these names stay handwritten until then because
 * they are transport concerns rather than payload types.
 */
export const ApiHeaders = {
  /** Carries a correlation ID through one logical operation. */
  correlationId: 'X-Correlation-Id',

  /** Carries the aggregate version for optimistic concurrency. */
  entityTag: 'ETag',

  /** Carries a client-supplied idempotency key for commands that must not execute twice. */
  idempotencyKey: 'Idempotency-Key',
} as const;

File: src/TouchlineManager.Contracts/Http/ApiHeaders.cs
namespace TouchlineManager.Contracts.Http;

/// <summary>
/// Custom HTTP header names that form part of the API contract.
/// </summary>
/// <remarks>
/// These live in Contracts because clients depend on them: the Angular PWA echoes the correlation
/// ID and quotes it in support requests, and the future native client does the same. Changing a name
/// here is a contract change.
/// </remarks>
public static class ApiHeaders
{
    /// <summary>
    /// Carries a correlation ID through one logical operation. Accepted on a request when it is
    /// safe, always returned on the response.
    /// </summary>
    public const string CorrelationId = "X-Correlation-Id";

    /// <summary>
    /// Carries the aggregate version for optimistic concurrency, and is required in
    /// <c>If-Match</c> on conflicting updates (ADR-0009).
    /// </summary>
    public const string EntityTag = "ETag";

    /// <summary>
    /// Carries a client-supplied idempotency key for commands that must not execute twice
    /// (ADR-0003, CONC-3).
    /// </summary>
    public const string IdempotencyKey = "Idempotency-Key";
}
